### DISCUSSION TITLE
Trivy error while walking the filesystem (timeout)

### DISCUSSION THREAD

**Discussion by AXDOOMER** [aquasecurity/trivy#4658] [Bugs]
### Description

Hello.

I'm trying to find which dependency file is causing an error. I'm using the `-d` flag on Trivy to get more information, but when Trivy times out, it doesn't tell me where it was stuck. (it really is stuck for long, increasing the timeout to 30 minutes gave it just enough time to finish)

This is an excerpt from the output: 

```
2023-06-16T17:29:49.118-0400    DEBUG   Resolving net.sf.jopt-simple:jopt-simple:5.0.2...
2023-06-16T17:29:49.134-0400    DEBUG   Start parent: org.sonatype.oss:oss-parent:7
2023-06-16T17:29:49.134-0400    DEBUG   Exit parent: org.sonatype.oss:oss-parent:7


2023-06-16T17:31:40.055-0400    WARN    Increase --timeout value
2023-06-16T17:31:40.055-0400    FATAL   filesystem scan error:
    github.com/aquasecurity/trivy/pkg/commands/artifact.Run
        /home/runner/work/trivy/trivy/pkg/commands/artifact/run.go:430
  - scan error:
    github.com/aquasecurity/trivy/pkg/commands/artifact.(*runner).scanArtifact
        /home/runner/work/trivy/trivy/pkg/commands/artifact/run.go:268
  - scan failed:
    github.com/aquasecurity/trivy/pkg/commands/artifact.scan
        /home/runner/work/trivy/trivy/pkg/commands/artifact/run.go:683
  - failed analysis:
    github.com/aquasecurity/trivy/pkg/scanner.Scanner.ScanArtifact
        /home/runner/work/trivy/trivy/pkg/scanner/scan.go:144
  - walk filesystem:
    github.com/aquasecurity/trivy/pkg/fanal/artifact/local.Artifact.Inspect
        /home/runner/work/trivy/trivy/pkg/fanal/artifact/local/fs.go:164
  - walk error:
    github.com/aquasecurity/trivy/pkg/fanal/walker.FS.walkFast
        /home/runner/work/trivy/trivy/pkg/fanal/walker/fs.go:90
  - unknown error with ./harbour/sailor/pocs/client/client-sdk/public/ubiservices/core/system/linux:
    github.com/aquasecurity/trivy/pkg/fanal/walker.NewFS.func1
        /home/runner/work/trivy/trivy/pkg/fanal/walker/fs.go:30
  - failed to analyze file:
    github.com/aquasecurity/trivy/pkg/fanal/walker.FS.Walk.func1
        /home/runner/work/trivy/trivy/pkg/fanal/walker/fs.go:66
  - analyze file (harbour/sailor/pocs/client/client-sdk/public/ubiservices/core/system/linux/atomic.inl):
    github.com/aquasecurity/trivy/pkg/fanal/artifact/local.Artifact.Inspect.func1
        /home/runner/work/trivy/trivy/pkg/fanal/artifact/local/fs.go:147
  - semaphore acquire:
    github.com/aquasecurity/trivy/pkg/fanal/analyzer.AnalyzerGroup.AnalyzeFile
        /home/runner/work/trivy/trivy/pkg/fanal/analyzer/analyzer.go:423
  - context deadline exceeded
  ```
  
 Is there a way to easily find out where it got stuck? I was testing it on about 450 repos. There must be a dependency file where there are performance issues scanning the file. 
  
  If I find where it gets stuck for long, I can do a proper bug report. 
  
  Thank you.

### Desired Behavior

Don't get stuck walking the file system

### Actual Behavior

I get stucks walking the file system

### Reproduction Steps

```bash
1.
2.
3.
...
```

### Target

Filesystem

### Scanner

None

### Output Format

None

### Mode

None

### Debug Output

```bash
See above
```

### Operating System

Ubuntu 22.04

### Version

```bash
0.42.1
```

### Checklist

- [ ] Run `trivy --reset`
- [ ] Read [the troubleshooting](https://aquasecurity.github.io/trivy/latest/docs/references/troubleshooting/)

**Comment by knqyf263** [comment]
Looks like it takes time to scan pom.xml. Does your pom.xml have a lot of dependencies?

**Reply by AXDOOMER** [reply]
I don't know. I opened this ticket and the reason was to see if the debug output could be improved so I would know on which file there was a timeout.

**Reply by knqyf263** [reply]
It is pom.xml

**Reply by AXDOOMER** [reply]
Indeed, but I've got over 200 projects with pom.xml files. I'll write a script to scan them one by one and find which ones are slow to scan. Is `org.sonatype.oss:oss-parent:7` supposed to be the name of the project?

**Reply by knqyf263** [reply]
>Is org.sonatype.oss:oss-parent:7 supposed to be the name of the project?

It is a parent POM.
